plugins/pulp_rpm/plugins/migrations/0031_yum_metadata_storage_path.py

```diff
@@ -0,0 +1,39 @@
+from gettext import gettext as _
+import logging
+import os
+import tempfile
+
+from pulp.server.db import model
+from pulp_rpm.plugins.db import models
+
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    Re-calculate storage path for yum_repo_metadata_file unit
+    """
+    metadata_files = models.YumMetadataFile.objects.filter()
+    repos = set()
+    for unit in metadata_files:
+        old_path = unit._storage_path
+        unit.set_storage_path(os.path.basename(unit._storage_path))
+        if unit._storage_path != old_path:
+            unit.save()
+            unit.safe_import_content(old_path)
+            repos_with_unit = model.RepositoryContentUnit.objects.filter(unit_id=unit.id)
```

Comment:
Since all you need is the ``repo_id``, and that is present on the unit, I don't think you need to do a DB query here. You can just do ``repos.add(unit.repo_id)``, and end up with a set of repo IDs. That would eliminate the list comprehension below as well.